src/main/java/io/zold/api/Transaction.java

```diff
@@ -37,4 +36,32 @@
      * @return The amount
      */
     long amount();
+
+    /**
+     * Smart transaction.
+     *
+     * @author Izbassar Tolegen (t.izbassar@gmail.com)
+     * @version $Id$
+     * @since 0.1
+     */
+    final class Smart implements Transaction {
+
+        /**
+         * Amount.
+         */
+        private final long amount;
+
+        /**
+         * Ctor.
+         * @param amount Amount
+         */
+        public Smart(final long amount) {
```

Comment:
@t-izbassar I think the constructor should accept `String`, where the supplied string represent one transaction line. Or at least we should have a constructor/method that does this.